src/foam/util/SafetyUtil.java

```diff
@@ -100,6 +100,42 @@ public static int compare(double o1, double o2) {
     return o1 == o2 ? 0 : o1 < o2 ? -1 : 1;
   }
 
+  public static int hash(Object o1) {
+    return java.util.Objects.hash(o1);
+  }
+
+  public static int hash(FObject o1) {
+    return java.util.Objects.hash(o1);
+  }
+
+  public static int hash(boolean o1) {
+    return java.util.Objects.hash(o1);
```

Comment:
This will convert it to an Object, which is what we were trying to avoid. Something like this:
return o1 ? 1 : 0;